### DISCUSSION TITLE
Request for Comment (RFC) archiving and discovery

### DISCUSSION THREAD

**Discussion by lmbr-pip** [o3de/o3de#6718] [General]
Hey O3DE community, 

I would like to understand and gather ideas about what we should be doing in relation to Request for Comments (RFCs) and archiving.

## The Current Process
* SIGs accepted an RFC and they archive it locally, some examples [SIG-Network](https://github.com/o3de/sig-network/tree/main/rfcs), [SIG-Core](https://github.com/o3de/sig-core/tree/main/rfcs), [SIG-Content)[https://github.com/o3de/sig-content/tree/main/rfcs]

## The Current Problems
* RFCs are 'hidden' in SIG archives, you have to know it exists and where to find it. 
     * Some SIGs are not archiving RFCs but only the issues. 
     * RFCs with cross-SIG impact are only kept in the main SIG repo (which makes it harder still).
* Theres no wider notification mechanisms that an RFC has been accepted, if you are lucky the SIG uses its mailing list/discord to highlight RFC acceptance.
* Theres no discovery mechanism, outside of GitHub general search (which is poor) - if you want to find prior RFCs in an area it can require a lot of leg work.
* Theres a confusing [o3de/rfcs](https://github.com/o3de/rfcs/) repro that is not used.

## Potential Solutions
* All SIGs follow a standard process to name RFCS  (this was proposed at O3DE inception but not followed).
    
Then?
* All SIGs archive accepted RFCs in o3de/rfcs  - Move to centralized ownership model, SIGs currently operate pretty independently. 
* SIGs archive metadata about accepted RFCs in o3de/rfcs - create document(s) that contain metadata (RFC number, title, summary and keywords) in o3de/rfcs that link to RFC archived in SIG repro.
* SIGs archive metadata (title, summary, keywords, links) on central page(s) on docs.o3de.org that link to archived RFC in SIG repros.
* Something else?
 
I tried looking at other open source processes like [Rust](https://github.com/rust-lang/rfcs) [Yarn](https://github.com/yarnpkg/rfcs) but these are based around a central RFC repo which is not a model we follow. I could not find a good example of a large project with many SIGs that has decentralized formal RFC archiving approach. Kubernetes has a similar structure, many decentralized SIGs, but they follow a much more informal issue based RFC process than we do (https://github.com/kubernetes-sigs/kind/issues?q=is%3Aissue+RFC) or I have not found the right pages yet.

**Comment by jeremyong** [comment]
I personally wouldn't be opposed to centralizing RFCs published in a single repo until volume necessitates finer subdivision. We can group RFCs by folder to organize, and this way, RFCs can easily reference other RFCs for technical designs that are cross-cutting. In the future, if the influx of RFCs is somehow large enough that this grouping becomes unwieldy, we could start grouping batches of RFCs by quarter, or by month.

What you lose with this organization is finer grained permissions, but I'm not sure this is an issue, given that all members of the community across different sigs contribute to a single root O3DE monorepo.

**Comment by mcphedar** [comment]
If SIG's want to keep their existing process (or even if they don't) we could use the new Github Project beta to create a board that groups the RFCs by SIG. The RFCs remain in their existing SIG repos, but are also added to a common project board.

The largest downside to this is that its a manual process. RFC guidelines would need to add this process to the end. It isn't overly complicated to explain and adding the RFC to the board takes seconds. That said, it is a manual process, which is a risk by itself.
